### ISSUE TITLE
`key-spacing.align` doesn't pay attention to non-whitespace before key

### ISSUE THREAD

**Issue by zyklus** [eslint/eslint#3267]
I'll admit my code style is weird, but `key-spacing.align` doesn't actually enforce alignment of values/colons like the documentation states, it enforces the alignment of keys _and_ values/colons.

If this is intended, the docs should be updated (and theoretically the error messages changed).  But I don't think it is:

```js
var x = {
  foo : 10
  , b : 20 // missing space after key
};

var x = {
  foo    : 10, // extra space after key
  /**/ b : 20  // missing space after key
};
```

these pass:

```js
var x = {
  foo : 10
  , b   : 20
};

var x = {
  foo : 10,
  /**/ b   : 20
};
```

what it "wants":

```js
var x = {
    foo : 10
  , b   : 20
};

var x = {
       foo : 10,
  /**/ b   : 20
};
```

eslint 0.24.1, `"key-spacing": { "align" : "colon" }`

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by btmills** [comment]
Right now the rule is checking the length of keys, which I just realized is a really inefficient way to do it, in addition to suffering cases like this. We'd be much better off checking the locations of tokens, either the colon (for `"align": "colon"`) or the token immediately after it (`"align": "value"`).

**Comment by nzakas** [comment]
@btmills so is this a bug?

**Comment by btmills** [comment]
In the case with leading commas, definitely - we should support that. Leading comments... potentially, but let's just fix both cases all at once with the token locations.
